community/kernel/src/main/java/org/neo4j/kernel/impl/factory/GraphDatabaseFacadeFactory.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.factory;
+
+import java.io.File;
+import java.util.Map;
+
+import org.neo4j.graphdb.config.Setting;
+import org.neo4j.graphdb.factory.GraphDatabaseSettings;
+import org.neo4j.helpers.Exceptions;
+import org.neo4j.helpers.Settings;
+import org.neo4j.kernel.AvailabilityGuard;
+import org.neo4j.kernel.extension.KernelExtensionFactory;
+import org.neo4j.kernel.impl.query.QueryEngineProvider;
+import org.neo4j.kernel.monitoring.Monitors;
+import org.neo4j.logging.LogProvider;
+import org.neo4j.logging.Logger;
+
+import static org.neo4j.helpers.Settings.ANY;
+import static org.neo4j.helpers.Settings.STRING;
+import static org.neo4j.helpers.Settings.illegalValueMessage;
+import static org.neo4j.helpers.Settings.matches;
+import static org.neo4j.helpers.Settings.setting;
+
+/**
+ * This is the main factory for creating database instances. It delegates creation to three different modules ({@link org.neo4j.kernel.impl.factory.PlatformModule}, {@link org.neo4j.kernel.impl.factory.EditionModule}, and {@link org.neo4j.kernel.impl.factory.DataSourceModule}),
+ * which create all the specific services needed to run a graph database.
+ *
+ * It is abstract in order for subclasses to specify their own {@link org.neo4j.kernel.impl.factory.EditionModule} implementations. Subclasses also have to set the edition name
+ * in overriden version of {@link #newFacade(java.util.Map, org.neo4j.kernel.impl.factory.GraphDatabaseFacadeFactory.Dependencies, GraphDatabaseFacade)}, which is used for logging and similar.
```

Comment:
Maybe it is better to define one more method `String getName()` instead of putting a requirement to modify `Map` parameter?